src/shared/error.ts

```diff
@@ -0,0 +1,40 @@
+import type { Show } from 'fp-ts/lib/Show'
+
+export const ResourceConflictError = (resourceIds: string[]) =>
+  ({
+    _tag: 'resource-conflict',
+    resourceIds
+  } as const)
+
+/**
+ *  Error when an id of a resource to be stored already exists in memory.
```

Comment:
```suggestion
 *  Error when an id of a resource to be stored already exists in the domain.
```

(memory is an implementation detail)

Reply:
Much better 👍